### PR TITLE
Move away from Busybox dhcp client to a more modern one

### PR DESCRIPTION
**Pull request by skarbat**
Small adaptation needed to migrate to the new DHCP client.
The new client will be working as an autonomous daemon already.
This PR complements: https://github.com/KanoComputing/kano-settings/pull/456

There are still things to address after a connection succeeds, as we have post-connect hooks.

### CODE REVIEW

**Comment by skarbat** [comment]
Tested the migration on an auto-upgraded image to scratch and seems to start working:

```
osupgrades@osupgrades ~ $ kano-test-dhcp
PASS: Sep 18 19:33:49 osupgrades logger[1593]: kano-dhcp-hook: REBOOT event ipaddr=192.168.1.130 on iface=wlan0
PASS: Sep 18 19:33:49 osupgrades logger[1595]: kano-dhcp-hook: launching network up scripts
PASS: Sep 18 19:33:49 osupgrades systemd[1]: Started /usr/bin/kano-sentry-startup REBOOT.
PASS: Sep 18 19:33:49 osupgrades systemd[1]: Started /usr/bin/kano-network-hook REBOOT.
PASS: Sep 18 19:33:49 osupgrades systemd[1]: Started /usr/bin/kano-set-system-date REBOOT.
PASS: Sep 18 19:33:49 osupgrades kano-sentry-startup[1599]: kano-sentry-startup [1599] info Ultimate parental control IS switched on
PASS: Sep 18 19:33:49 osupgrades systemd[1]: Started /usr/bin/kano-dashboard-sysupdates bound 192.168.1.130.
PASS: Sep 18 19:34:06 osupgrades systemd[1]: Time has been changed
PASS: Sep 18 19:34:06 osupgrades kano-set-system-date[1621]: kano-set-system-date [1621] info rdate SUCCESS: Mon Sep 18 19:34:06 CEST 2017
failures=0
```

File: dhcpcd-hooks/60-kano-dhcp-debug [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#
+# Copyright (C) 2017 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
+#
+# Simple script to diagnose dhcpcd events.
+# To enable, edit /etc/systemd/journalctl.conf and set Storage=persistent
+#
+
+ipaddr=`ip addr show wlan0 | grep inet | awk '{print $2}' | sed -s 's/\/.*//')`
+logger "kano-dhcp-debug: $reason ifup? $if_up interface? $interface carrier? $ifcarrier wireless? $ifwireless ssid=$ifssid ipaddr=$ipaddr"
```
Comment (tombettany) [inline dhcpcd-hooks/60-kano-dhcp-debug]: Will this not execute and do nothing regardless of whether the the logging is enabled? Is there any way to fully disable this when the logging isn't enabled?

File: dhcpcd-hooks/60-kano-dhcp-debug [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#
```
Comment (tombettany) [inline dhcpcd-hooks/60-kano-dhcp-debug]: Do these `dhcpcd-hooks` require a shebang?

Comment (skarbat) [reply dhcpcd-hooks/60-kano-dhcp-debug]: Nope, they are sourced by `dhcpcd-run-hooks`

 * https://manpages.debian.org/testing/dhcpcd5/dhcpcd-run-hooks.8.en.html

File: dhcpcd-hooks/60-kano-dhcp-debug (hunk 1 above)
Comment (skarbat) [reply dhcpcd-hooks/60-kano-dhcp-debug]: That is true, and I think I'll ship this file under `/usr/share` instead.

**Pull request merged**